### PR TITLE
Update 'http:' to 'https:'

### PR DESCRIPTION
**Pull request by CanOfBees**
Sorry - this should have been updated before.

## Interested Parties ##
@markpbaggett

### CODE REVIEW

File: oai-mods-add-thumbnail-url.xsl [hunk 1]
```diff
@@ -41,13 +41,13 @@
                 <xsl:attribute name="usage">
                     <xsl:value-of select="'primary display'"/>
                 </xsl:attribute>
-                <xsl:value-of select="following::mods:identifier[starts-with(.,'http://')]"/>
+                <xsl:value-of select="following::mods:identifier[starts-with(.,'https://')]"/>
```
Comment (markpbaggett) [inline oai-mods-add-thumbnail-url.xsl]: Is it me, or does it make more sense to do this:

```
<xsl:value-of select="following::mods:identifier[starts-with(.,'http')]"/>
```

This seems like a safer bet, but maybe it doesn't matter.  Thoughts?  If you disagree, I'll approve, just want to ask first.

File: oai-mods-add-thumbnail-url.xsl [hunk 2]
```diff
@@ -41,13 +41,13 @@
...
             </xsl:element>
             <xsl:element name="url">
                 <xsl:attribute name="access">
                     <xsl:value-of select="'preview'"/>
                 </xsl:attribute>
-                <xsl:value-of select="concat(following::mods:identifier[starts-with(.,'http://')],'/datastream/TN/view')"/>
+                <xsl:value-of select="concat(following::mods:identifier[starts-with(.,'https://')],'/datastream/TN/view')"/>
```
Comment (markpbaggett) [inline oai-mods-add-thumbnail-url.xsl]: Same as above:

```
<xsl:value-of select="concat(following::mods:identifier[starts-with(.,'http')],'/datastream/TN/view')"/>
```

Thoughts?

File: oai-mods-add-thumbnail-url.xsl [hunk 3]
```diff
@@ -60,7 +60,7 @@
         there's a test to make sure that we don't add a second
         mods:location if there's already one.
     -->
-    <xsl:template match="/mods:mods/mods:identifier[starts-with(.,'http://')]">
+    <xsl:template match="/mods:mods/mods:identifier[starts-with(.,'https://')]">
```
Comment (markpbaggett) [inline oai-mods-add-thumbnail-url.xsl]: Same as above:

```
<xsl:template match="/mods:mods/mods:identifier[starts-with(.,'https')]">
```

Thoughts?

**Review by markpbaggett — COMMENTED** [review]
I don't feel strongly about this, but it feels like if we would have just done this from the beginning we wouldn't have to change anything.  Not requesting changes, but am requesting thoughts / an acknowledgement.  If you feel like changing, if not, I'll approve.  If you have any banter to add, I'd love to hear it.

**Review by markpbaggett — CHANGES_REQUESTED** [review]
Also, I know this is out of scope of the review, but this whole template looks scuffed to me.

Check the template on line 30 that starts with:

```
    <xsl:template match="mods:location[not(parent::mods:relatedItem)]">
        <xsl:copy>
            <xsl:apply-templates select="@*|node()"/>
            <!--
                constructing elements to avoid copied namespaces.
                also, i'm a bit lazy.
            -->
            <xsl:element name="url">
                <xsl:attribute name="access">
                    <xsl:value-of select="'object in context'"/>
                </xsl:attribute>
                <xsl:attribute name="usage">
                    <xsl:value-of select="'primary display'"/>
                </xsl:attribute>
                <xsl:value-of select="following::mods:identifier[starts-with(.,'http')]"/>
            </xsl:element>
            <xsl:element name="url">
                <xsl:attribute name="access">
                    <xsl:value-of select="'preview'"/>
                </xsl:attribute>
                <xsl:value-of select="concat(following::mods:identifier[starts-with(.,'http')],'/datastream/TN/view')"/>
            </xsl:element>
        </xsl:copy>
    </xsl:template>
```

It's basically saying, let's find a location node that doesn't live in a relatedItem and construct some urls.

Then, check the very next template:

```
<xsl:template match="/mods:mods/mods:identifier[starts-with(.,'http://')]">
        <xsl:choose>
            <!-- do we already have a top-level mods:location? -->
            <xsl:when test="/mods:mods/mods:location"/>
            <!-- if we don't, then this test runs -->
            <xsl:when test="not(/mods:mods/mods:location)">
                <xsl:element name="location">
                    <xsl:element name="url">
                        <xsl:attribute name="access">
                            <xsl:value-of select="'object in context'"/>
                        </xsl:attribute>
                        <xsl:attribute name="usage">
                            <xsl:value-of select="'primary display'"/>
                        </xsl:attribute>
                        <xsl:value-of select="."/>
                    </xsl:element>
                    <xsl:element name="url">
                        <xsl:attribute name="access">
                            <xsl:value-of select="'preview'"/>
                        </xsl:attribute>
                        <xsl:value-of select="concat(.,'/datastream/TN/view')"/>
                    </xsl:element>    
                </xsl:element>                
            </xsl:when>
        </xsl:choose>
    </xsl:template>
```

There is some weird stuff here.  The context of the match is oddly root. I'm not sure what the data looks like before this is ran, so I'm not sure that matters.  Is it an OAI-PMH response, or just straight MODS.  

It seems like the point of this is to create a location element if one does not exist but identifiers do.  I'm not sure if this is still a thing, but the template is still here.  I'd say we should change the match to:

```
<xsl:template match="/mods:mods/mods:identifier[starts-with(.,'http')]">
```

It makes sense to do that here if you're changing all these http based matches. Or remove the template.

File: oai-mods-add-thumbnail-url.xsl (hunk 1 above)
Comment (CanOfBees) [reply oai-mods-add-thumbnail-url.xsl]: I definitely don't see a problem with this.

File: oai-mods-add-thumbnail-url.xsl (hunk 2 above)
Comment (CanOfBees) [reply oai-mods-add-thumbnail-url.xsl]: Ditto.

File: oai-mods-add-thumbnail-url.xsl (hunk 3 above)
Comment (CanOfBees) [reply oai-mods-add-thumbnail-url.xsl]: Ditto.

**Review by markpbaggett — APPROVED** [review]

**Pull request merged**